synapseclient/models/mixins/table_operator.py

```diff
@@ -1898,174 +1918,196 @@ async def main():
+                        ) or not isna(cell_value):
+                            partial_change_values[
+                                column_id
+                            ] = _convert_pandas_row_to_python_types(
+                                cell=cell_value, column_type=column_type
+                            )
+                        else:
+                            partial_change_values[column_id] = None
+
+                if partial_change_values != {}:
+                    total_row_count_updated += 1
+                    partial_change = PartialRow(
+                        row_id=row.ROW_ID,
+                        etag=row_etag,
+                        values=[
+                            {
+                                "key": partial_change_key,
+                                "value": partial_change_value,
+                            }
+                            for partial_change_key, partial_change_value in partial_change_values.items()
+                        ],
                     )
-
-                if len(matching_row[column].values) == 0:
-                    continue
-                column_id = self.columns[column].id
-                column_type = self.columns[column].column_type
-                cell_value = matching_row[column].values[0]
-                if cell_value != row[column]:
-                    if (
-                        isinstance(cell_value, list) and len(cell_value) > 0
-                    ) or not isna(cell_value):
-                        partial_change_values[
-                            column_id
-                        ] = _convert_pandas_row_to_python_types(
-                            cell=cell_value, column_type=column_type
-                        )
-                    else:
-                        partial_change_values[column_id] = None
-
-            if partial_change_values != {}:
-                partial_chage = PartialRow(
-                    row_id=row_id,
-                    etag=row_etag,
-                    values=[
-                        {"key": partial_change_key, "value": partial_change_value}
-                        for partial_change_key, partial_change_value in partial_change_values.items()
-                    ],
+                    rows_to_update.append(partial_change)
+                    indexs_of_original_df_with_changes.append(matching_row.index[0])
+
+        if not dry_run and rows_to_update:
+            chunked_rows_to_update = []
+            chunk_size_limit = 1.5 * 1024 * 1024  # 1.5MB
+            current_chunk_size = 0
+            chunk = []
+            for row in rows_to_update:
+                row_size = sys.getsizeof(row.to_synapse_request())
+                if current_chunk_size + row_size > chunk_size_limit:
+                    chunked_rows_to_update.append(chunk)
+                    chunk = []
+                chunk.append(row)
+                current_chunk_size += row_size
```

Comment:
This is some of the largest changes here. This interface that we use to submit a `PartialRow` change is what limits requests to 2MB or less:

https://rest-docs.synapse.org/rest/POST/entity/id/table/transaction/async/start.html

In this case I am using 1.5MB to allow for some overhead room.